cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableService.java

```diff
@@ -0,0 +1,449 @@
+                                  "Only text and csv are supported as native formats");
+      String schema = FileSetProperties.getExploreSchema(properties);
+      Preconditions.checkNotNull(schema, "for native formats, explore schema must be given in dataset properties");
+      String delimiter = null;
+      if ("text".equals(format)) {
+        delimiter = FileSetProperties.getExploreFormatProperties(properties).get("delimiter");
+      } else if ("csv".equals(format)) {
+        delimiter = ",";
+      }
+      return createStatementBuilder.setSchema(schema)
+        .setRowFormatDelimited(delimiter, null)
+        .buildWithFileFormat("TEXTFILE");
+    } else {
+      // format not given, look for serde, input format, etc.
+      String serde = FileSetProperties.getSerDe(properties);
+      String inputFormat = FileSetProperties.getExploreInputFormat(properties);
+      String outputFormat = FileSetProperties.getExploreOutputFormat(properties);
+
+      Preconditions.checkArgument(serde != null && inputFormat != null && outputFormat != null,
+                                  "All of SerDe, InputFormat and OutputFormat must be given in dataset properties");
+      return createStatementBuilder.setRowFormatSerde(serde)
+        .buildWithFormats(inputFormat, outputFormat);
+    }
+  }
+
+  private String generateDeleteStatement(String name) {
+    return String.format("DROP TABLE IF EXISTS %s", getHiveTableName(name));
+  }
+
+  private String generateHivePartitionKey(PartitionKey key) {
+    StringBuilder builder = new StringBuilder("(");
+    String sep = "";
+    for (Map.Entry<String, ? extends Comparable> entry : key.getFields().entrySet()) {
+      String fieldName = entry.getKey();
+      Comparable fieldValue = entry.getValue();
+      String quote = fieldValue instanceof String ? "'" : "";
+      builder.append(sep).append(fieldName).append("=").append(quote).append(fieldValue.toString()).append(quote);
+      sep = ", ";
+    }
+    builder.append(")");
+    return builder.toString();
+  }
+
+  /**
+   * Given a record-enabled dataset, its record type and generate a schema string compatible with Hive.
+   *
+   * @param dataset The data set
+   * @return the hive schema
+   * @throws UnsupportedTypeException if the dataset is neither RecordScannable, nor RecordWritable,
+   * or if the row type is not a record or contains null types.
+   */
+  private String hiveSchemaFor(Dataset dataset) throws UnsupportedTypeException {
+    if (dataset instanceof RecordScannable) {
+      return hiveSchemaFor(((RecordScannable) dataset).getRecordType());
+    } else if (dataset instanceof RecordWritable) {
+      return hiveSchemaFor(((RecordWritable) dataset).getRecordType());
+    }
+    throw new UnsupportedTypeException("Dataset neither implements RecordScannable not RecordWritable.");
+  }
+
+  // TODO: replace with SchemaConverter.toHiveSchema when we tackle queries on Tables.
```

Comment:
Should there be a jira here? Not sure if this was introduced in this PR or was just moved around.